pkg/deploy/strategy/recreate/recreate.go

```diff
@@ -88,6 +93,25 @@ func (s *RecreateDeploymentStrategy) Deploy(from *kapi.ReplicationController, to
 	return s.DeployWithAcceptor(from, to, desiredReplicas, nil)
 }
 
+// recordControllerWarnings records the replication controller warnings into a
+// deployment event sink.
+func (s *RecreateDeploymentStrategy) recordControllerWarnings(rc *kapi.ReplicationController) {
```

Comment:
> @kargakis that check is for tests otherwise they panic (we don't init the sink for unit tests...)

Oh, you are using the recorder for emitting the event on the config. Do you need the recorder since you have the client? Can't you just create the event?